examples/ActionGroup/ActionGroup.Tests.ps1

```diff
@@ -2,31 +2,69 @@ BeforeAll {
   Import-Module Az.InfrastructureTesting
 }
 
+$resourceType = "ActionGroup"
```

Comment:
In order to put it in the BeforeAll block without making PSScriptAnalyzer mad, I just found out that you'll need to preface the variable name with `$Script::`

Reply:
After testing this further `$Script::variable` does not work in Pester, or at least I couldn't get it to work. I don't know why, but it throws `RuntimeException: The property '' cannot be found on this object. Verify that the property exists and can be set.`, which makes absolutely no sense (at least none that I can find yet).

The IDE will warn of a PSScriptAnalyzer error without the `$Script::` in front of the variable because it thinks that it is scoped to the `BeforeAll` block, but it is not because of the way Pester works.

If this passes PSScriptAnalyzer in the PR pipeline, with just adding the variables to the `BeforeAll` then that's the likely way forward.

Also I found this:

https://pester.dev/docs/migrations/v4-to-v5

Which clearly states that for Pester tests `Put all your test-code into It, BeforeAll, BeforeEach, AfterAll or AfterEach. Put no test-code directly into Describe, Context or on the top of your file, without wrapping it in one of these blocks, unless you have a good reason to do so.`

Reply:
It's official, it fails PSScriptAnalyzer when the variables are put inside the `BeforeAll` block.

We might need to bring this up at design session because I see two options: go against Pester best practices and define the variables where they are not intended to be defined or go against PSScriptAnalyzer best practices and define them where Pester wants them defined.

Reply:
FYI - there is something similar working in `Modules\BenchPress.Azure\Tests\Public\Confirm-Resource.Tests.ps1`. It uses `$script:variableName` in the `BeforeEach` block to define a fake `ConfirmResult` that can be reused throughout the tests. Just to confirm, this same approach doesn't work in the `BeforeAll` block?